internal/provisioner/objects/deployment/deployment_test.go

```diff
@@ -270,3 +271,34 @@ func TestNodePlacementDeployment(t *testing.T) {
 	checkDeploymentHasNodeSelector(t, deploy, selectors)
 	checkDeploymentHasTolerations(t, deploy, tolerations)
 }
+
+func TestDesiredDeploymentWhenSettingDisabledFeature(t *testing.T) {
+	testCases := []struct {
+		description      string
+		disabledFeatures []contour_v1.Feature
+	}{
+		{
+			description:      "several disabled features",
+			disabledFeatures: []contour_v1.Feature{"ns1", "ns2"},
```

Comment:
nit: use a feature name here just for readability (functionally doesnt matter and could by any string, but just for someone reading it later)